sys/kern/kern_cheri_revoke.c

```diff
@@ -313,21 +313,21 @@ kern_cheri_revoke(struct thread *td, int flags,
 	vm = vmspace_acquire_ref(td->td_proc);
 	vmm = &vm->vm_map;
 
-	/*
-	 * We need some value that's more or less "now", but we don't have
-	 * to be too picky about it.  This value is not reported to
-	 * userland, just used to guide the interlocking below.
-	 */
-	if ((flags & CHERI_REVOKE_IGNORE_START) != 0) {
-		start_epoch = cheri_revoke_st_get_epoch(
-		    vmm->vm_cheri_revoke_st);
-	}
-
 	/* Serialize and figure out what we're supposed to do */
 	vm_map_lock(vmm);
 	{
 		int ires = 0;
 
+		/*
+		 * We need some value that's more or less "now", but we don't have
+		 * to be too picky about it.  This value is not reported to
+		 * userland, just used to guide the interlocking below.
+		 */
+		if ((flags & CHERI_REVOKE_IGNORE_START) != 0) {
+			start_epoch = cheri_revoke_st_get_epoch(
+			    vmm->vm_cheri_revoke_st);
+		}
+
```

Comment:
We probably should just drop `IGNORE_START`, too; it's not _that_ hard for allocators to do a trivial version of the segmented quarantine thing and so always have some epoch value at hand.

Reply:
Ok. It's not very intrusive so I thought I'd leave it alone for now. Removing it would also require me to update the cheribsdtest cases which use it; I'm not sure yet whether they actually need that flag.